src/dialog/dlgreplacecuecolor.cpp

```diff
@@ -0,0 +1,321 @@
+        queryString += QStringLiteral(" WHERE ") + queryStringConditions.join(QStringLiteral(" AND "));
+    }
+
+    // Select affected queues
+    QSqlQuery query(database);
+    query.prepare(queryString);
+    for (auto i = queryValues.constBegin(); i != queryValues.constEnd(); i++) {
+        query.bindValue(i.key(), i.value());
+    }
+
+    if (!query.exec()) {
+        LOG_FAILED_QUERY(query);
+        return {};
+    }
+    QMap<int, int> ids;
+    int idColumn = query.record().indexOf("id");
+    int trackIdColumn = query.record().indexOf("track_id");
+    while (query.next()) {
+        ids.insert(query.value(idColumn).toInt(), query.value(trackIdColumn).toInt());
+    }
+
+    return ids;
+}
+
+void DlgReplaceCueColor::slotDatabaseIdsSelected() {
+    QMap<int, int> ids = m_dbSelectFuture.result();
+
+    if (ids.size() == 0) {
+        QMessageBox::warning(this, tr("No colors changed!"), tr("No cues matched the specified criteria."));
+        setApplyButtonEnabled(true);
+        return;
+    }
+
+    QSet<int> cueIds;
+    QSet<TrackId> trackIds;
+    for (auto i = ids.constBegin(); i != ids.constEnd(); i++) {
+        cueIds << i.key();
+        trackIds << TrackId(i.value());
+    }
+
+    if (QMessageBox::question(
+                this,
+                tr("Really replace colors?"),
+                tr("Really replace the colors of %1 cues in %2 tracks? This change cannot be undone!").arg(QString::number(cueIds.size()), QString::number(trackIds.size()))) == QMessageBox::No) {
+        setApplyButtonEnabled(true);
+        return;
+    }
+
+    mixxx::RgbColor::optional_t newColor = mixxx::RgbColor::fromQString(pushButtonNewColor->text());
+    VERIFY_OR_DEBUG_ASSERT(newColor) {
+        setApplyButtonEnabled(true);
+        return;
+    }
+
+    m_dbUpdateFuture = QtConcurrent::run(
+            this,
+            &DlgReplaceCueColor::updateCues,
+            cueIds,
+            trackIds,
+            *newColor);
+    m_dbUpdateFutureWatcher.setFuture(m_dbUpdateFuture);
+}
+
+void DlgReplaceCueColor::updateCues(QSet<int> cueIds, QSet<TrackId> trackIds, mixxx::RgbColor newColor) {
+    // The pooler limits the lifetime of all thread-local connections. It will
+    // be closed by its destructor when it goes out of scope
+    const mixxx::DbConnectionPooler dbConnectionPooler(m_pDbConnectionPool);
+    QSqlDatabase database = mixxx::DbConnectionPooled(m_pDbConnectionPool);
+
+    //Open the database connection in this thread.
+    VERIFY_OR_DEBUG_ASSERT(database.isOpen()) {
+        qWarning() << "Failed to open database for cue color replace dialog."
+                   << database.lastError();
+        return;
+    }
+
+    //Give thread a low priority
+    QThread* thisThread = QThread::currentThread();
+    thisThread->setPriority(QThread::LowPriority);
+
+    ScopedTransaction transaction(database);
+    QSqlQuery query(database);
+    query.prepare("UPDATE " CUE_TABLE " SET color=:color WHERE id=:id");
```

Comment:
Race condition: The color might have changed already since it is been read.

Updates must only succeed if the old color is still present, i.e. a WHERE condition for the old color is missing.

I suggest to invoke this function with a list of update records and the new color. Each update record contains:
- The track id
- A list of pairs with cue id, and old color

QSet<TrackId> is collected from queries with an update count of 1. A query with an update count of 0 is ignored and should not abort the operation.